packages/types/src/data/abstract-query.ts

```diff
@@ -0,0 +1,240 @@
+}
+
+export interface AbstractQueryFieldNodeRelatedBase {
+	fieldNodes: AbstractQueryFieldNode[];
+
+	modifiers?: AbstractQueryModifiers;
+}
+
+export type AbstractQueryFieldNodeRelated =
+	| AbstractQueryFieldNodeRelatedManyToOne
+	| AbstractQueryFieldNodeRelatedOneToMany
+	| AbstractQueryFieldNodeRelatedAnyToOne
+	| AbstractQueryFieldNodeRelatedOneToAny;
+
+interface AbstractQueryFieldNodeRelatedJoinMany {
+	current: {
+		fields: string[];
+	};
+
+	external: {
+		datastore?: string;
+		collection: string;
+		fields: string[];
+	};
+}
+
+interface AbstractQueryFieldNodeRelatedJoinAny {
+	current: {
+		collectionField: string;
+		fields: string[];
+	};
+
+	external: {
+		datastore?: string;
+		fields: string[];
+	};
+}
+
+export interface AbstractQueryFieldNodeRelatedManyToOne extends AbstractQueryNode, AbstractQueryFieldNodeRelatedBase {
+	type: 'm2o';
+
+	join: AbstractQueryFieldNodeRelatedJoinMany;
+}
+
+export interface AbstractQueryFieldNodeRelatedOneToMany extends AbstractQueryNode, AbstractQueryFieldNodeRelatedBase {
+	type: 'o2m';
+
+	join: AbstractQueryFieldNodeRelatedJoinMany;
+}
+
+export interface AbstractQueryFieldNodeRelatedAnyToOne extends AbstractQueryNode, AbstractQueryFieldNodeRelatedBase {
+	type: 'a2o';
+
+	join: AbstractQueryFieldNodeRelatedJoinAny;
+}
+
+export interface AbstractQueryFieldNodeRelatedOneToAny extends AbstractQueryNode, AbstractQueryFieldNodeRelatedBase {
+	type: 'o2a';
+
+	join: AbstractQueryFieldNodeRelatedJoinAny;
+}
+
+// =================================================================================================
+
+/**
+ * Optional attributes to customize the query results
+ * @typeParam sort - Specifies the order of the results
+ * @typeParam limit - Specifies the maximum amount of returning results
+ */
+export interface AbstractQueryModifiers {
+	limit?: AbstractQueryNodeLimit;
+	offset?: AbstractQueryNodeOffset;
+	sort?: AbstractQueryNodeSort;
+	filter?: AbstractQueryNodeLogical | AbstractQueryNodeCondition;
+}
+
+/**
+ * Specifies the maximum amount of returning results
+ * @typeParam type - set to 'limit'
+ * @typeParam value - the limit value
+ */
+interface AbstractQueryNodeLimit {
+	type: 'limit';
+	value: number;
+}
+
+/**
+ * Specifies the number of items to skip before returning results
+ * @typeParam type - set to 'offset'
+ * @typeParam value - the offset value
+ */
+interface AbstractQueryNodeOffset {
+	type: 'offset';
+	value: number;
+}
+
+/**
+ * Specifies the order of the results
+ * @typeParam type - set to 'sort'
+ * @typeParam direction - 'ascending' or 'descending'
+ * @typeParam target - the node on which the sorting should be applied
+ */
+interface AbstractQueryNodeSort {
+	type: 'sort';
+	direction: 'ascending' | 'descending';
+	target:
+		| AbstractQueryFieldNodePrimitive
+		| AbstractQueryFieldNodeFn
+		| AbstractQueryFieldNodeRelatedManyToOne
+		| AbstractQueryFieldNodeRelatedAnyToOne;
+}
+
+/**
+ * Used to create logical operations
+ * @typeParam type - set to 'sort'
```

Comment:
```suggestion
 * @typeParam type - set to 'logical'
```

Reply:
on Friday we decided to not use the `@typeParam` and instead put the comment above the attribute. I moved the remaining comments down and corrected the things you mentioned. we also agreed to not do comments when it's obvious so I also removed some comments.